src/signals.rs

```diff
@@ -3,30 +3,72 @@ use crossbeam_channel::{unbounded, Receiver};
 use signal_hook::consts::signal::*;
 use signal_hook::iterator::Signals;
 
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc,
+};
+
 use crate::thread::spawn;
 
-pub(crate) enum Signal {
-    Exit,
-    Trigger,
+#[derive(Clone)]
+pub(crate) struct ExitFlag {
+    flag: Arc<AtomicBool>,
 }
 
-pub(crate) fn register() -> Receiver<Signal> {
-    let ids = [
-        SIGINT, SIGTERM,
-        SIGUSR1, // allow external triggering (e.g. via bitcoind `blocknotify`)
-    ];
-    let (tx, rx) = unbounded();
-    let mut signals = Signals::new(&ids).expect("failed to register signal hook");
-    spawn("signal", move || {
-        for id in &mut signals {
-            info!("notified via SIG{}", id);
-            let signal = match id {
-                SIGUSR1 => Signal::Trigger,
-                _ => Signal::Exit,
-            };
-            tx.send(signal).context("failed to send signal")?;
+impl ExitFlag {
+    fn new() -> Self {
+        ExitFlag {
+            flag: Arc::new(AtomicBool::new(false)),
         }
-        Ok(())
-    });
-    rx
+    }
+
+    pub fn get(&self) -> bool {
+        self.flag.load(Ordering::Relaxed)
+    }
+
+    fn set(&self) {
```

Comment:
It's not entirely clear from the method names whether it means "exiting" or "running" (although type name could be considered sufficient?) Perhaps name them `is_exiting()` and `set_exiting()`

Reply:
Renamed `get` into `is_set`, keeping `set` as is - since it's private.